cpp/modmesh/buffer/pymod/wrap_SimpleArrayPlex.cpp

```diff
@@ -53,13 +53,161 @@ class MODMESH_PYTHON_WRAPPER_VISIBILITY WrapSimpleArrayPlex : public WrapBase<Wr
                     { return wrapped_type(make_shape(shape), datatype); }),
                 pybind11::arg("shape"),
                 pybind11::arg("dtype"))
+            .def_timed(
+                pybind11::init(&init_array_plex_with_value),
+                pybind11::arg("shape"),
+                pybind11::arg("value"),
+                pybind11::arg("dtype"))
+            .def(
+                pybind11::init(
+                    [](pybind11::array & arr_in)
+                    {
+                        shape_type shape;
+                        for (ssize_t i = 0; i < arr_in.ndim(); ++i)
+                        {
+                            shape.push_back(arr_in.shape(i));
+                        }
+                        std::shared_ptr<ConcreteBuffer> const buffer = ConcreteBuffer::construct(
+                            arr_in.nbytes(),
+                            arr_in.mutable_data(),
+                            std::make_unique<ConcreteBufferNdarrayRemover>(arr_in));
+                        return wrapped_type(shape, buffer, pybind11::str(arr_in.dtype()));
+                    }),
+                pybind11::arg("array"))
             .def_property_readonly("typed", &get_typed_array)
             /// TODO: should have the same interface as WrapSimpleArray
             ;
     }
 
-    /// return the typed function from the arrayplex
-    static pybind11::object get_typed_array(wrapped_type const & array_plex)
+    /// Initialize the arrayplex with the given value
+    /// NOLINTNEXTLINE(bugprone-easily-swappable-parameters)
+    static wrapped_type init_array_plex_with_value(pybind11::object const & shape_in, pybind11::object const & value, std::string const & datatype)
+    {
+        const shape_type shape = make_shape(shape_in);
+        wrapped_type array_plex(shape, datatype);
+
+        // cast value to correct type based on the datatype
+        switch (array_plex.data_type())
+        {
+        case DataType::Bool:
+        {
+            if (!pybind11::isinstance<pybind11::bool_>(value))
```

Comment:
The repeating error-checking code probably should use a macro to simplify.  Could you please take a look?  If it is an obviously good idea, you can try it in a later PR.

Reply:
For now, maybe maintaining the current state is suitable. I think that employing macros doesn't enhance readability unless we encounter excessive repetition of the same code.